gui/src/app/pages/HomePage/ImportWindow.tsx

```diff
@@ -0,0 +1,112 @@
+import { FieldsContentsMap, FileNames, FileRegistry, mapFileContentsToModel } from "../../SPAnalysis/FileMapping"
+import { SPAnalysisContext } from "../../SPAnalysis/SPAnalysisContextProvider"
+import { deserializeZipToFiles, parseFile } from "../../SPAnalysis/SPAnalysisSerialization"
+import UploadFilesArea from "./UploadFilesArea"
+
+type ImportWindowProps = {
+    onClose: () => void
+}
+
+const ImportWindow: FunctionComponent<ImportWindowProps> = ({ onClose }) => {
+    const { update } = useContext(SPAnalysisContext)
+    const [errorText, setErrorText] = useState<string | null>(null)
+    const [filesUploaded, setFilesUploaded] = useState<{name: string, content: ArrayBuffer}[] | null>(null)
+    const [showReplaceProjectOptions, setShowReplaceProjectOptions] = useState<boolean>(false)
+
+    const importUploadedFiles = useCallback(async (o: {replaceProject: boolean}) => {
+        const {replaceProject} = o
+        if (!filesUploaded) return
+        try {
+            if ((filesUploaded.length === 1) && (filesUploaded[0].name.endsWith('.zip'))) {
+                // a single .zip file
+                const fileManifest = await deserializeZipToFiles(filesUploaded[0].content)
+                update({ type: 'loadFiles', files: fileManifest, clearExisting: replaceProject })
+            }
+            else if ((filesUploaded.length === 1) && (filesUploaded[0].name.endsWith('.stan'))) {
+                // a single .stan file
+                if (replaceProject) {
+                    update({ type: 'retitle', title: filesUploaded[0].name })
+                }
+                const fileManifest: Partial<FieldsContentsMap> = { 'stanFileContent': parseFile(filesUploaded[0].content) }
+                update({ type: 'loadFiles', files: fileManifest, clearExisting: replaceProject })
+            }
+            else {
+                const files: Partial<FileRegistry> = {}
+                for (const file of filesUploaded) {
+                    if ( !Object.values(FileNames).includes(file.name as any) ){
+                        throw Error(`Unrecognized file: ${file.name}`)
+                    }
+                    files[file.name as FileNames] = parseFile(file.content)
+                }
+
+                const fileManifest = mapFileContentsToModel(files);
+                update({ type: 'loadFiles', files: fileManifest, clearExisting: replaceProject })
+            }
+            onClose()
+        }
+        catch (e: any) {
+            setErrorText(e.message)
+        }
+    }, [filesUploaded, onClose, update])
+
+    useEffect(() => {
+        if (!filesUploaded) return
+        if ((filesUploaded.length === 1) && (!filesUploaded[0].name.endsWith('.zip'))) {
+            // The user has uploaded a single file and it is not a zip file. In
+            // this case we want to give the user the option whether or not to
+            // replace the current project.
+            setShowReplaceProjectOptions(true)
+        }
```

Comment:
Happy to leave this until after merge, just echoing my discussion from #53 that I think uploading a single file should keep the drop box enabled to allow you to also add a second/third etc